### DISCUSSION TITLE
Eliminate global current epoch counter

### DISCUSSION THREAD

**Discussion by Alan-Jowett** [microsoft/ebpf-for-windows#701] [Ideas]
@nibanks had an interesting suggestion that is worth exploring. Currently modifying and reading [Current Epoch](https://github.com/microsoft/ebpf-for-windows/blob/d8f00f851382ba5b3ef574a479ab70aa80bf05c8/libs/platform/ebpf_epoch.c#L71) is expensive.

Is there some scheme that can be employed to move this to a per-CPU value?

Requirements of the CE:
1) It has to be a non-decreasing value.
2) The value read by the [free code](https://github.com/microsoft/ebpf-for-windows/blob/d8f00f851382ba5b3ef574a479ab70aa80bf05c8/libs/platform/ebpf_epoch.c#L299) must be the same or higher than the value read at the start of any currently executing RCU regions.

**Comment by dthaler** [comment]
PR #699 was one part of this investigation.
